File: Utilities/Models/Hall.cs

using System.ComponentModel.DataAnnotations.Schema;

namespace Utilities.Models;

public class Hall : BaseModel
{
    [Required]
    public int? Row { get; set; }
    [Required]
    public int? Column { get; set; }

    [ForeignKey("Id")] // Dette angiver navnet på fremmednøglen
    public Guid? CinemaId { get; set; }

    [JsonIgnore]
    public Cinema? Cinema { get; set; }   
    public ICollection<Movie>? Movies { get; set; } = new List<Movie>();
}

File: Client/src/app/Models/hall.ts
import { BaseModel, IBaseModel } from './baseModel';
import { ICinema } from './cinema';
import { IMovie } from './movie';

export interface IHall extends IBaseModel {
  row?: number;
  column?: number;
  movies?: IMovie[];
  cinemas?: ICinema[];
}

export class Hall extends BaseModel implements IHall {
  row?: number;
  column?: number;
  movies?: IMovie[];
  cinemas?: ICinema[];

  constructor() {
    super();
    this.movies = [];
    this.cinemas = [];
  }
}
